preprocess/download_so.py

```diff
@@ -28,21 +21,35 @@ def reporthook(blocknum, blocksize, totalsize):
         sys.stderr.write("read %d\n" % (readsofar,))
 
 
-if uname.system == "Darwin":
-    if uname.machine == "arm64":
-        # TODO Download my own build of libcurl-impersonate for M1 Mac
-        url = ""
-    else:
-        url = f"https://github.com/yifeikong/curl-impersonate/releases/download/v{VERSION}/libcurl-impersonate-v{VERSION}.{uname.machine}-macos.tar.gz"
-elif uname.system == "Windows":
-    url = f"https://github.com/yifeikong/curl-impersonate/releases/download/v{VERSION}/libcurl-impersonate-v{VERSION}.x86_64-win32.tar.gz"
+_uname = platform.uname()
+system = _uname.system
+machine = _uname.machine
+
+if system == "Windows":
+    libdir = "./lib"
+elif system == "Darwin" and machine == "x86_64":
+    libdir = "/Users/runner/work/_temp/install/lib"
 else:
-    url = f"https://github.com/yifeikong/curl-impersonate/releases/download/v{VERSION}/libcurl-impersonate-v{VERSION}.{uname.machine}-linux-gnu.tar.gz"
-
-filename = "./curl-impersonate.tar.gz"
-if url:
-    print(f"Download libcurl-impersonate-chrome from {url}")
-    urlretrieve(url, filename, (None if os.getenv("GITHUB_ACTIONS") else reporthook))
-    shutil.unpack_archive(filename, LIBDIR)
-    if uname.system == "Windows":
-        shutil.copy2(f"{LIBDIR}/libcurl.dll", "curl_cffi")
+    libdir = "/usr/local/lib"
+
+if len(sys.argv) == 3 and sys.argv[2] == '1':  # i686 32-bit flag
```

Comment:
Checking sys.argv with more than one arguments is kind of code smell, but let's leave it as is for now, we should make the names more consistent in curl-impersonate release files and move the code in download_so.py to Makefile in the future.

Reply:
Correct, tracked in #197